lib/sentry/logger_backend.ex

```diff
@@ -128,27 +97,54 @@ defmodule Sentry.LoggerBackend do
     :ok
   end
 
-  defp init(config, %__MODULE__{} = state) do
-    level = Keyword.get(config, :level, state.level)
+  defp log(level, msg, meta, state) do
+    sentry = meta[:sentry] || get_sentry_from_callers(meta[:callers] || [])
 
-    include_logger_metadata =
-      Keyword.get(config, :include_logger_metadata, state.include_logger_metadata)
+    opts =
+      [
+        event_source: :logger,
+        extra: %{logger_metadata: logger_metadata(meta, state), logger_level: level},
+        result: :none
+      ] ++ Map.to_list(sentry)
+
+    case meta[:crash_reason] do
+      {exception, stacktrace} when is_list(stacktrace) ->
+        Sentry.capture_exception(exception, [stacktrace: stacktrace] ++ opts)
+
+      reason when is_atom(reason) and not is_nil(reason) ->
+        Sentry.capture_exception(reason, opts)
 
-    ignore_plug = Keyword.get(config, :ignore_plug, state.ignore_plug)
+      _ ->
+        if state.capture_log_messages do
+          try do
+            if is_binary(msg), do: msg, else: :unicode.characters_to_binary(msg)
+          rescue
+            _ -> :ok
+          else
+            msg -> Sentry.capture_message(msg, opts)
+          end
+        end
+    end
+  end
 
-    %{
-      state
-      | level: level,
-        include_logger_metadata: include_logger_metadata,
-        ignore_plug: ignore_plug
-    }
+  defp get_sentry_from_callers([head | tail]) do
```

Comment:
```suggestion
  defp get_sentry_from_callers([head | tail]) when is_pid(head) do
```

Otherwise a list of `callers: [1]` would make it fail.

Reply:
We should probably add a test for malformed callers. :)